services/templates.ts

```diff
@@ -266,3 +292,139 @@ export const formatTemplatesWithPriceData = (
+    });
+
+    const userAssetsWithHiddenByTemplateId = {};
+    accountResponseWithHidden.data.templates.map(({ assets, template_id }) => {
+      userAssetsWithHiddenByTemplateId[template_id] = assets;
+    });
+
+    const templateIds = accountResponse.data.templates
+      .map(({ template_id }) => template_id)
+      .splice((page - 1) * 10, 10);
+
+    if (!templateIds.length) return [];
+
+    const templates = await getTemplatesFromTemplateIds(templateIds);
+
+    const lowestPricesByTemplateId = await getLowestPricesForAllCollectionTemplates(
+      { type: DEFAULT_COLLECTION }
+    );
+
+    const templatesWithAssetsForSaleCount = formatTemplatesWithLowPriceAndAssetCount(
+      {
+        templateIds: templateIds,
+        templates: templates,
+        assetCountById: userAssetsByTemplateId,
+        assetCountByIdWithHidden: userAssetsWithHiddenByTemplateId,
+        lowPriceById: lowestPricesByTemplateId,
+      }
+    );
+    return templatesWithAssetsForSaleCount;
+  } catch (e) {
+    throw new Error(e);
+  }
+};
+
+/**
+ * Function to add total asset count, assets for sale, and lowest price to template data for each template
+ * Used in conjunction with function getTemplatesWithUserAssetCount
+ * @param templateIds list of templateIds of templates to add data to
+ * @param templates   templates of the template Ids listed in templateIds param
+ * @param assetCountById  total number of assets for each template that user owns
+ * @param assetCountByIdWithHidden  total number of assets for each template that user owns minus those currently offered for sale
+ * @param lowPriceById  lowest price of asset currently on offer for each template
+ * @returns {Template[]}
+ */
+
+const formatTemplatesWithLowPriceAndAssetCount = ({
+  templateIds,
+  templates,
+  assetCountById,
+  assetCountByIdWithHidden,
+  lowPriceById,
+}: formatTemplatesWithLowPriceAndAssetCountProps) => {
+  const templatesWithAssetsForSaleCount = templateIds.map((templateId) => {
+    const template = templates.find(({ template_id }) => {
+      return templateId == template_id;
+    });
+    if (template) {
+      template.totalAssets = `${assetCountById[templateId]}`;
+
+      const assetsForSale =
+        parseInt(assetCountById[templateId]) -
+        parseInt(assetCountByIdWithHidden[templateId]);
+
+      template.assetsForSale = `${assetsForSale}`;
+      template.lowestPrice = lowPriceById[templateId];
+    }
+    return template;
+  });
+  return templatesWithAssetsForSaleCount;
+};
+
+/**
+ * Function to get templates using an array of tempalte ids as reference
+ * @param templateIds templatesIds to grab templates for
+ * @returns {Template[]}
+ */
+
+export const getTemplatesFromTemplateIds = async (
+  templateIds: string[]
+): Promise<Template[]> => {
+  try {
+    const templatesQueryObject = {
+      symbol: TOKEN_SYMBOL,
+      collection_name: DEFAULT_COLLECTION,
+      ids: `${templateIds.join(',')}`,
```

Comment:
Small nitpick - can we remove the `${}` interpolation since joining the array of `templateIds` will always return a string?